gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -359,6 +374,13 @@ public void submitFlowToDagManager(FlowSpec flowSpec, Dag<JobExecutionPlan> jobE
     try {
       // Send the dag to the DagManager
       this.dagManager.get().addDag(jobExecutionPlanDag, true, true);
+
+      /*
+      Adhoc flows can be deleted after persisting it in DagManager as the DagManager's failure recovery method ensures
+      it will be executed in the event of downtime. Note that the responsibility of the multi-active scheduler mode ends
+      after this method is completed AND the consumption of a launch type event is committed to the consumer.
+       */
+      deleteSpecFromCatalogIfAdhoc(flowSpec);
```

Comment:
If I remember correctly, they should already be getting deleted from the flow catalog

Reply:
To confirm, did you check the callers of `FlowCatalog` `remove` method ? Is it missing some case?